include/libdnf/module/module_profile.hpp

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_MODULE_MODULE_PROFILE_HPP
+#define LIBDNF_MODULE_MODULE_PROFILE_HPP
+
+#include <memory>
+#include <string>
+#include <vector>
+
+struct _ModulemdProfile;
+
+
+namespace libdnf::module {
+
+
+// @replaces libdnf:module:modulemd/ModuleProfile.hpp:class:ModuleProfile
+class ModuleProfile {
+public:
+    /// @return The profile name.
+    /// @since 5.0
+    //
+    // @replaces libdnf:module:modulemd/ModuleProfile.hpp:method:ModuleProfile.getName()
+    std::string get_name() const;
+
+    /// @return The profile description.
+    /// @since 5.0
+    //
+    // @replaces libdnf:module:modulemd/ModuleProfile.hpp:method:ModuleProfile.getDescription()
+    std::string get_description() const;
+
+    // TODO(pkratoch): Describe the format of the RPM strings in the docstring (is it NEVRAs?).
+    /// @return The list of RPMs.
+    /// @since 5.0
+    //
+    // @replaces libdnf:module:modulemd/ModuleProfile.hpp:method:ModuleProfile.getContent()
+    std::vector<std::string> get_content() const;
+
+    /// @return `true` if the profile is default for the associated stream.
+    /// @since 5.0
+    //
+    // @replaces libdnf:module:modulemd/ModuleProfile.hpp:method:ModuleProfile.isDefault()
+    bool is_default() const;
+
+    // @replaces libdnf:module:modulemd/ModuleProfile.hpp:ctor:ModuleProfile.ModuleProfile()
+    ModuleProfile() : profile(nullptr) {}
```

Comment:
IMO it's a good practice to initialize the members in their declaration (here it would be `_ModulemdProfile * profile{nullptr};` at the end of class declaration). That way you make sure you haven't forgotten the initialization in some constructor. And here it saves you the default constructor definition.

Reply:
I believe that if it will be possible we should delete the default costructor. I suggest that it was created only because SWIG required it for vector costructor.